applications/FluidDynamicsApplication/custom_elements/distance_smoothing_element.cpp

```diff
@@ -0,0 +1,710 @@
+                            solid_normal = geometry[i].FastGetSolutionStepValue(NORMAL);
+                            const double norm = Kratos::norm_2(solid_normal);
+#ifdef KRATOS_DEBUG
+                            KRATOS_WARNING_IF("DistanceSmoothingElement", norm < 1.0e-12) << "WARNING: Normal close to zero" <<std::endl;
+#endif
+                            solid_normal = (1.0/norm)*solid_normal;
+                        }
+
+                        temp_value += face_shape_func(i)*Kratos::inner_prod(solid_normal, grad_phi_old[i]);
+                    }
+
+                    for (unsigned int i = 0; i < num_nodes; i++){
+                        tempBCRHS[i] += epsilon * temp_value * face_weight * face_shape_func(i);
+                    }
+                }
+            }
+
+        }
+    }
+
+    noalias(rLeftHandSideMatrix) = tempM + tempA;
+    noalias(rRightHandSideVector) = tempBCRHS + prod(tempM,phi_old) - prod(rLeftHandSideMatrix,phi_dof);
+
+    KRATOS_CATCH("");
+}
+
+/**
+ * ELEMENTS inherited from this class have to implement next
+ * CalculateLocalSystem, CalculateLeftHandSide and CalculateRightHandSide methods
+ * they can be managed internally with a private method to do the same calculations
+ * only once: MANDATORY
+ */
+
+/**
+ * this is called during the assembling process in order
+ * to calculate all elemental contributions to the global system
+ * matrix and the right hand side
+ * @param rLeftHandSideMatrix: the elemental left hand side matrix
+ * @param rRightHandSideVector: the elemental right hand side
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+template< >
+void DistanceSmoothingElement<3>::CalculateLocalSystem(
+    MatrixType& rLeftHandSideMatrix,
+    VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    constexpr static unsigned int num_dim  = 3;
+    constexpr static unsigned int num_nodes  = num_dim + 1;
+    constexpr static unsigned int num_faces = num_nodes;
+    constexpr static unsigned int num_face_nodes = num_nodes - 1;
+
+    const double dt = rCurrentProcessInfo.GetValue(DELTA_TIME);
+    const auto& geometry = this->GetGeometry();
+    const double he = ElementSizeCalculator<num_dim,num_nodes>::AverageElementSize(geometry);
+    const double epsilon = 1.0e2*dt*he*he;
+
+    BoundedMatrix<double,num_nodes,num_dim> DN_DX;  // Gradients matrix
+    array_1d<double,num_nodes> N; // dimension = number of nodes . Position of the gauss point
+
+    array_1d<double,num_nodes> phi_dof; //dimension = number of DOFs, needed since we are using a residualbased approach
+    array_1d<double,num_nodes> phi_old; //dimension = number of DOFs
+    array_1d<VectorType,num_nodes> grad_phi_old;
+
+    BoundedMatrix<double,num_nodes,num_nodes> tempM = ZeroMatrix(num_nodes,num_nodes);
+    BoundedMatrix<double,num_nodes,num_nodes> tempA = ZeroMatrix(num_nodes,num_nodes);
+
+    array_1d<double,num_nodes> tempBCRHS = ZeroVector(num_nodes);
```

Comment:
please take a look at the style guide in the future

Reply:
Sure. I always have problem with naming matrices!
I will keep them as they are for now.